1_fetch/src/my_happy_script.R

```diff
@@ -87,10 +112,45 @@ render_data <- list(pgdl_980mean = filter(eval_data, model_type == 'pgdl', exper
                     pb_100mean = filter(eval_data, model_type == 'pb', exper_id == "similar_100") %>% pull(rmse) %>% mean %>% round(2),
                     pgdl_2mean = filter(eval_data, model_type == 'pgdl', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2),
                     pb_2mean = filter(eval_data, model_type == 'pb', exper_id == "similar_2") %>% pull(rmse) %>% mean %>% round(2))
-
+     
 template_1 <- 'resulted in mean RMSEs (means calculated as average of RMSEs from the five dataset iterations) of {{pgdl_980mean}}, {{dl_980mean}}, and {{pb_980mean}}°C for the PGDL, DL, and PB models, respectively.
   The relative performance of DL vs PB depended on the amount of training data. The accuracy of Lake Mendota temperature predictions from the DL was better than PB when trained on 500 profiles 
   ({{dl_500mean}} and {{pb_500mean}}°C, respectively) or more, but worse than PB when training was reduced to 100 profiles ({{dl_100mean}} and {{pb_100mean}}°C respectively) or fewer.
   The PGDL prediction accuracy was more robust compared to PB when only two profiles were provided for training ({{pgdl_2mean}} and {{pb_2mean}}°C, respectively). '
 
-whisker.render(template_1 %>% str_remove_all('\n') %>% str_replace_all('  ', ' '), render_data ) %>% cat(file = file.path(project_output_dir, 'model_diagnostic_text.txt'))
+# since this is inside the dataRender function, this should be able to use the local variable render_data
+whisker.render(template_1 %>% str_remove_all('\n') %>% str_replace_all('  ', ' '), render_data) %>% cat(file = file.path(textOutput, 'model_diagnostic_text.txt'))
+}
+
+########
+# DIRECTIONS :
+# run this code block after adding each function to the functions in your global environement:
+
+# obtain data from URL:
+# create new output directory
+Visualize <- '3_Visualize'
+visualize_output_dir <- '3_Visualize/out'   
+visualize_src <- '3_Visualize/src'   
+process_output_dir <- '2_process/out'
+dir.create(visualize_output_dir)
+dir.create(process_output_dir)
+dir.create(Visualize)
+dir.create(visualize_src)
```

Comment:
If you were using github and collaborating with someone on this code, you wouldn't need to create a directory each time because in many cases they would already exist. For example, any `src` directory that contains committed code would already exist. In some cases, you might not be git tracking data (like the csv you downloaded), so you may want to `dir.create` the out folders. I think you could simplify the above code chunk quite a bit. You're not using `Visualize` anywhere, so can probably delete line 131. You can delete 133 and 138 because those filepaths will exist once you commit the code to that place.